File: python/10-null-spiral/src/room_lookup.py
class RoomLookup:
    def __init__(self):
        pass

    def get_room_for_student(self, student_id, db):
        if db is not None:
            student = db.get_student(student_id)
            if student is not None:
                enrollment = db.get_enrollment(student["id"])
                if enrollment is not None:
                    course = db.get_course(enrollment["courseId"])
                    if course is not None:
                        section = db.get_section(course["defaultSectionId"])
                        if section is not None:
                            room = db.get_room(section["roomId"])
                            if room is not None:
                                return room["name"]
                            else: return None
                        else: return None
                    else: return None
                else: return None
            else: return None
        else: return None

File: golang/10-null-spiral/room_lookup.go
package kata

type Student struct{ ID int }
type Enrollment struct{ CourseID int }
type Course struct{ DefaultSectionID int }
type Section struct{ RoomID int }
type Room struct{ Name string }
type DB interface {
	GetStudent(int) *Student
	GetEnrollment(int) *Enrollment
	GetCourse(int) *Course
	GetSection(int) *Section
	GetRoom(int) *Room
}

type RoomLookup struct{}

func NewRoomLookup() *RoomLookup {
	return &RoomLookup{}
}

func (rl *RoomLookup) get_room_for_student(studentID int, db DB) *string {
	if db != nil {
		student := db.GetStudent(studentID)
		if student != nil {
			enrollment := db.GetEnrollment(student.ID)
			if enrollment != nil {
				course := db.GetCourse(enrollment.CourseID)
				if course != nil {
					section := db.GetSection(course.DefaultSectionID)
					if section != nil {
						room := db.GetRoom(section.RoomID)
						if room != nil {
							return &room.Name
						} else {
							return nil
						}
					} else {
						return nil
					}
				} else {
					return nil
				}
			} else {
				return nil
			}
		} else {
			return nil
		}
	} else {
		return nil
	}
}
